src/main/java/com/synopsys/integration/jenkins/JenkinsProxyHelper.java

```diff
@@ -40,34 +40,53 @@
-        if (StringUtils.isNotBlank(proxyConfig.getUserName())) {
-            final String[] possiblyDomainSlashUsername = proxyConfig.getUserName().split(Pattern.quote("\\"));
+    private String url;
+    private String proxyHost;
+    private int proxyPort;
+    private String proxyUsername;
+    private String proxyPassword;
+    private List<Pattern> ignoredProxyHosts;
+    private String ntlmDomain;
+    private String ntlmWorkstation;
+    private ProxyInfo proxyInfo = ProxyInfo.NO_PROXY_INFO;
+
+    public JenkinsProxyHelper(final ProxyConfiguration proxyConfiguration, final String url) {
+        this.url = url;
+
+        this.proxyHost = proxyConfiguration.name;
+        this.proxyPort = proxyConfiguration.port;
+        this.proxyPassword = proxyConfiguration.getPassword();
+        this.ignoredProxyHosts = proxyConfiguration.getNoProxyHostPatterns();
+        this.ntlmWorkstation = StringUtils.EMPTY;
+
+        if (StringUtils.isNotBlank(proxyConfiguration.getUserName())) {
+            final String[] possiblyDomainSlashUsername = proxyConfiguration.getUserName().split(Pattern.quote("\\"));
             if (possiblyDomainSlashUsername.length == 1 || possiblyDomainSlashUsername[0].length() == 0) {
-                ntlmDomain = null;
-                username = proxyConfig.getUserName();
+                this.ntlmDomain = null;
+                this.proxyUsername = proxyConfiguration.getUserName();
             } else {
-                ntlmDomain = possiblyDomainSlashUsername[0];
-                username = possiblyDomainSlashUsername[1];
+                this.ntlmDomain = possiblyDomainSlashUsername[0];
+                this.proxyUsername = possiblyDomainSlashUsername[1];
             }
+        } else {
+            this.ntlmDomain = null;
+            this.proxyUsername = null;
         }
-
-        return getProxyInfo(url, proxyConfig.name, proxyConfig.port, username, proxyConfig.getPassword(), proxyConfig.getNoProxyHostPatterns(), ntlmDomain, StringUtils.EMPTY);
     }
 
-    public static ProxyInfo getProxyInfo(final String url, final String proxyHost, final int proxyPort, final String proxyUsername, final String proxyPassword, final List<Pattern> ignoredProxyHosts, final String ntlmDomain,
-        final String ntlmWorkstation) {
-        ProxyInfo proxyInfo = ProxyInfo.NO_PROXY_INFO;
+    public JenkinsProxyHelper(final String url, final String proxyHost, final int proxyPort, final String proxyUsername, final String proxyPassword, final List<Pattern> ignoredProxyHosts, final String ntlmDomain,
+        String ntlmWorkstation) {
+        this.url = url;
+        this.proxyHost = proxyHost;
+        this.proxyPort = proxyPort;
+        this.proxyUsername = proxyUsername;
+        this.proxyPassword = proxyPassword;
+        this.ignoredProxyHosts = ignoredProxyHosts;
+        this.ntlmDomain = ntlmDomain;
+        this.ntlmWorkstation = ntlmWorkstation;
+    }
 
+    public ProxyInfo getProxyInfo() {
```

Comment:
I believe this method should be the api of the JenkinsProxyHelper class-- rather than constructing the JenkinsProxyHelper with a ProxyConfiguration and a url, this method should take both as method parameters.

This would allow an instance of JenkinsProxyHelper to be used for multiple urls, instead of just one.

Reply:
I found myself rewriting SynopsysCredentialsHelper today-- https://github.com/synopsys-sig/jenkins-common/pull/8 might be the API we want for the helpers. Construct them with a Jenkins instance, and then they know how to talk to it to get what they want.